src/OrchardCore.Modules/OrchardCore.Sms.Azure/Services/AzureSmsProviderBase.cs

```diff
@@ -0,0 +1,79 @@
+using Azure.Communication.Sms;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Sms.Azure.Models;
+
+namespace OrchardCore.Sms.Azure.Services;
+
+public abstract class AzureSmsProviderBase : ISmsProvider
+{
+    private readonly AzureSmsOptions _providerOptions;
+    private readonly IPhoneFormatValidator _phoneFormatValidator;
+    private readonly ILogger _logger;
+
+    protected readonly IStringLocalizer S;
+
+    public AzureSmsProviderBase(
+        AzureSmsOptions options,
+        IPhoneFormatValidator phoneFormatValidator,
+        ILogger logger,
+        IStringLocalizer stringLocalizer)
+    {
+        _providerOptions = options;
+        _phoneFormatValidator = phoneFormatValidator;
+        _logger = logger;
+        S = stringLocalizer;
+    }
+
+    public abstract LocalizedString Name { get; }
+
+    public virtual async Task<SmsResult> SendAsync(SmsMessage message)
+    {
+        ArgumentNullException.ThrowIfNull(message);
+
+        if (!_providerOptions.IsEnabled)
+        {
+            return SmsResult.Failed(S["The Azure SMS Provider is disabled."]);
+        }
+
+        _logger.LogDebug("Attempting to send SMS to {Recipent}.", message.To);
+
+        if (string.IsNullOrWhiteSpace(message.To))
+        {
+            return SmsResult.Failed(S["The phone number for the recipent is required.", message.To]);
+        }
+        else
```

Comment:
You can reduce indentation here by removing the else {} since you are returning on the if condition above.